crates/oxc_parser/src/lexer/number.rs

```diff
@@ -83,72 +82,121 @@ fn parse_binary(s: &str) -> f64 {
     result as f64
 }
 
+// ==================================== OCTAL ====================================
+
+/// b'0' is 0x30 and b'7' is 0x37.
+///
+/// So we can convert from any octal digit to its value with `c & 7`.
+/// This is produces more compact assembly than `c - b'0'`.
+///
+/// <https://godbolt.org/z/9rYTsMoMM>
+const fn octal_byte_to_value(c: u8) -> u8 {
+    debug_assert!(c >= b'0' && c <= b'7');
+    c & 7
+}
 #[allow(clippy::cast_precision_loss, clippy::cast_lossless)]
 fn parse_octal(s: &str) -> f64 {
-    // b'0' is 0x30 and b'7' is 0x37.
-    // So we can convert from any octal digit to its value with `c & 7`.
-    // This is produces more compact assembly than `c - b'0'`.
-    // https://godbolt.org/z/9rYTsMoMM
-    const fn byte_to_value(c: u8) -> u8 {
-        debug_assert!(c >= b'0' && c <= b'7');
-        c & 7
-    }
-    #[cfg(test)]
-    {
-        static_assertions::const_assert_eq!(byte_to_value(b'0'), 0);
-        static_assertions::const_assert_eq!(byte_to_value(b'7'), 7);
+    const MAX_FAST_OCTAL_LEN: usize = 21;
```

Comment:
Please add a comment as to why 21 is the "magic number" here.

Reply:
Done